packages/dds/task-manager/src/taskManager.ts

```diff
@@ -454,8 +431,12 @@ export class TaskManagerClass
 		} else if (!this.connected) {
 			// If we are disconnected (and attached), wait to be connected and submit volunteer op
 			disconnectHandler();
-		} else if (!this.assigned(taskId) && !this.queued(taskId)) {
-			submitVolunteerOp();
+		} else if (!this.assigned(taskId) && !this.queuedOptimistically(taskId)) {
+			const latestPendingOp = this.latestPendingOps.get(taskId);
+			if (latestPendingOp?.type !== "volunteer") {
+				// We don't need to send a second volunteer op if we just sent one.
+				submitVolunteerOp();
+			}
```

Comment:
If I'm assigned according to consensus, but have a pending abandon op, I should probably also send a volunteer op here?  Does this all simplify down to a queuedOptimistically check?
```suggestion
		} else if (!this.queuedOptimistically(taskId)) {
			// We don't need to send a second volunteer op if we just sent one.
			submitVolunteerOp();
```